File: problems/algorithms/array-roman-number/js/response/roman-number.js
function toRoman(number) {
  const romanMatrix = [
    [1000, 'M'],
    [900, 'CM'],
    [500, 'D'],
    [400, 'CD'],
    [100, 'C'],
    [90, 'XC'],
    [50, 'L'],
    [40, 'XL'],
    [10, 'X'],
    [9, 'IX'],
    [5, 'V'],
    [4, 'IV'],
    [1, 'I'],
  ];

  let result = [];

  for (let romanInfo of romanMatrix) {
    let count = parseInt(number / romanInfo[0]);
    number = number % romanInfo[0];
    if (count) result.push(romanInfo[1].repeat(count));
  }

  return result.join('');
}

export { toRoman };

File: src/pages/exercises/array-roman-number/codes/ruby/response/src/roman_number.rb
def toRoman(number)
  romanMatrix = {
    "M" => 1000,
    "CM" => 900,
    "D" => 500,
    "CD" => 400,
    "C" => 100,
    "XC" => 90,
    "L" => 50,
    "XL" => 40,
    "X" => 10,
    "IX" => 9,
    "V" => 5,
    "IV" => 4,
    "I" => 1
  }
  result = []
  array = romanMatrix.to_a
  array.each do |romanInfo|
    count = Integer(number / romanInfo[1])
    number = number % romanInfo[1]
    while count > 0
      result.push(romanInfo[0])
      count = count - 1
    end
  end
  return result.join("")
end
